### DISCUSSION TITLE
Request: Configurable Option to disable "Pre-Movement Chat" soft-muting.

### DISCUSSION THREAD

**Discussion by RubyRezal** [GriefPrevention/GriefPrevention#953] [Ideas]
As a server administrator, I would like the ability to totally disable the "Pre-Movement Chat" soft-muting.

I run a private community server where active online players are welcoming new people in chat. I've noticed multiple new and returning players getting flagged by this feature when all they are doing is responding to a friendly welcome from other online players.  

I feel soft-muting players who perform "Pre-Movement Chat" is an overly aggressive, unwelcoming, and undesirable feature in my environment. Players should not be punished for saying "Hello" or responding to other players welcoming them upon first logging in. I would like to completely disable this feature of GriefPrevention's chat moderation functionality on my server.

Related issue: #511

Optional questions:

**Is your feature request related to a problem? Please describe.**
Yes. GriefPrevention is aggressively muting players that are behaving perfectly reasonably for a community environment.

**Describe alternatives you've considered**
Without an option to disable this feature my only alternative would be to remove GriefPrevention and find another plugin. I'm deeply reluctant to do this as the ability to create and resize claims without commands is what attracted me to this plugin and I've not found a comparable alternative.

I am also using CoreProtect as a fallback so I can rollback griefing of unclaimed space but I would prefer not to rely on it exclusively or have to switch to a different land claiming plugin.

Thank you.

**Comment by RoboMWM** [comment]
Before, when achievements were a thing, we used that to determine if a player has actually played on the server to warrant getting a free pass for the movement requirement. You used to be able to set this achievement via the spigot.yml. For similar functionality we've resorted to using the "picked up" statistic for wood logs to emulate this since advancements are mutable. If you have ideas on how to improve this functionality I would be interested.

If the pre-movement chat isn't already tied to the antispam functionality then that should be fixed.

**Comment by RubyRezal** [comment]
Here's my scenerio: A friend logs into my server for the first time, players online welcome them, they say "Hello!" and are instantly soft-muted, nobody can hear them respond except admins.

This is a totally private server, I'm not concerned about a new player logging in and chatting before doing anything else.  I do want spam protection just in case, but I don't want my server to require people to leave spawn and pickup x amount of anything before they can say "Hello". It's way too restrictive for the open and friendly environment we want in our MC world.

My suggestion for improvement is the ability to optionally disable the penalty for "Pre-Movement Chat" without turning spam protection off entirely. A setting in config and/or command to selectively disable "Pre-Movement Chat" automatic soft-mute would be most appreciated.

**Comment by RoboMWM** [comment]
If there's a way to use the pre-existing vanilla mechanics, that'd be preferable instead of adding yet another config option. E.g., you could "force" everyone to have already acquired the "Getting Wood" achievement pre 1.13

**Comment by RubyRezal** [comment]
We are running 1.16.1.

What if I add a plugin to my server that just automatically gives players a starter kit with enough wood to disable Pre-Movement Chat soft-mute?

If that would work I can probably add that to my server as a workaround.

**Comment by RubyRezal** [comment]
We are running Paper 131, Minecraft version 1.16.1.

If there is a way to automatically give all players the prerequisite to allow them to chat immediately upon their first log-in I'm willing to give that a go. I just need to know specifically what is required.

Specific guidance on how to achieve my desired result of disabling Pre-Movement Chat Soft-Mute for all players without a specific option in GriefPrevention to do so would be appreciated.

To be totally honest, it's kind of frustrating to have to work around this issue using methods outside of GriefPrevention instead of having an option to disable this within GriefPrevention its self.

**Comment by RoboMWM** [comment]
I'm assuming you use the rest of the antispam features?

**Comment by RoboMWM** [comment]
The answer will determine how I'll implement this feature and what release it'll be in.
